src/index.ts

```diff
@@ -222,6 +222,11 @@ export type UseAsyncReturn<
   currentParams: Args | null;
 };
 
+const isPromise = (promise: unknown): promise is Promise<unknown> =>
+  (typeof promise === 'object' || typeof promise === 'function') &&
+  !!promise &&
+  typeof (promise as Promise<unknown>).then === 'function';
```

Comment:
thanks

can you explain a bit this solution?

Have you seen the same solution used by another open-source tool?
It's worth commenting what's going on here imho

Reply:
Added comments explaining the function

Reply:
Ideally `Promise` type should be change to `PromiseLike` everywhere to ensure no method other than `then` is used. `then` is the only method required for promise implementation to be Promises/A+ compliant.

Reply:
@slorber this is the same check that [`is-promise`](https://github.com/then/is-promise/blob/master/index.js) uses.

I would recommend using `PromiseLike` for this guard instead of `Promise` for the reason mentioned above and [here](https://github.com/then/is-promise/issues/39).

If using `PromiseLike` requires type changes elsewhere, I'd be happy to do the work in a follow up PR :)

Reply:
Changed `isPromise` to `isPromiseLike` and made required type changes